### PR TITLE
Initial setup of controller and reconciler for VerrazzanoManagedCluster objects

### PR DESCRIPTION
**Pull request by mgianatagh**
The initial set of a controller and reconciler for VerrazzanoManagedCluster resources.  The reconciler does not do any real work yet.

Added integration tests for testing the creating and deletion of VerrazzanoManagedCluster resources.

Added unit tests for the new controller.

### CODE REVIEW

**Review by markxnelson — APPROVED** [review]

File: platform-operator/controllers/verrazzanomanagedcluster/verrazzanomanagedcluster_controller.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2021, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
```
Comment (pfmackin) [inline platform-operator/controllers/verrazzanomanagedcluster/verrazzanomanagedcluster_controller.go:2]: You should only have one controller per controllers/xyz directory, so you don’t need such a long name for the go file.

File: platform-operator/go.mod [hunk 2]
```diff
@@ -1,11 +1,12 @@
-// Copyright (c) 2020, Oracle and/or its affiliates.
+// Copyright (c) 2020, 2021, Oracle and/or its affiliates.
 // Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
 
 module github.com/verrazzano/verrazzano/platform-operator
 
 go 1.13
 
 require (
+	github.com/go-logr/logr v0.1.0
```
Comment (bobdonat) [inline platform-operator/go.mod:9]: this is an extra dependency which is not required.  the zap logger is the only logger needed.

File: platform-operator/main.go [hunk 3]
```diff
@@ -146,6 +147,15 @@ func main() {
 		mgr.GetWebhookServer().CertDir = config.CertDir
 	}
 
+	// Setup the reconciler for VerrazzanoManagedCluster objects
+	if err = (&clusterscontroller.VerrazzanoManagedClusterReconciler{
+		Client: mgr.GetClient(),
+		Log:    ctrl.Log.WithName("controllers").WithName("VerrazzanoManagedCluster"),
```
Comment (bobdonat) [inline platform-operator/main.go:153]: the "Log" field is not needed and if used you will not get consistent logging with the caller field.

File: platform-operator/go.mod (hunk 2 above)
Comment (mgianatagh) [reply platform-operator/go.mod:9]: fixed in next PR

File: platform-operator/main.go (hunk 3 above)
Comment (mgianatagh) [reply platform-operator/main.go:153]: removed in next PR

File: platform-operator/controllers/verrazzanomanagedcluster/verrazzanomanagedcluster_controller.go (hunk 1 above)
Comment (mgianatagh) [reply platform-operator/controllers/verrazzanomanagedcluster/verrazzanomanagedcluster_controller.go:2]: I was following the pattern I saw in the application-operator folders containing controllers.  Are you suggesting to simplify the name to "controller.go"?

**Pull request merged**